WordPress/gutenberg
# Refactor/toolbar button component to typescript

Part of #35744.

_**To be merged after https://github.com/WordPress/gutenberg/pull/47087.**_

## What?

Convert the `ToolbarButton` component to TypeScript.

## Why?

This PR is needed as part of an effort to convert `Toolbar` and related `Toolbar` subcomponents to TypeScript.

## How?

By adding types to `ToolbarButton` and `ToolbarButtonContainer`.

## Testing Instructions

1. Create or otherwise open a post
2. Add a block that would have some toolbar controls (e.g. `Image`)
3. Expected: Toolbar buttons should fundamentally work the same as before:
